File: ws/BankingDotNetCore/src/BankingDotNetCore/Server/IAccount.cs
namespace BankingDotNetCore.Server
{
    public interface IAccount
    {
        void accrueInterest();
        void deposit(float amount);
        void withdraw(float amount);
        float Balance { get; set; }
        string Name { get; set; }
        int Number { get; set; }
        float PositiveInterest { get; set; }
        float NegativeInterest { get; set; }
    }
}

File: rmi/BankingForRMI/src/dk/sdu/mmmi/opn/banking/server/IAccount.java
package dk.sdu.mmmi.opn.banking.server;

import java.rmi.Remote;
import java.rmi.RemoteException;

public interface IAccount extends Remote {

	public abstract float getBalance() throws RemoteException;

	public abstract String getName() throws RemoteException;

	public abstract void deposit(float amount) throws RemoteException;

	public abstract void withdraw(float amount) throws RemoteException;

	public abstract int getNumber() throws RemoteException;

	public abstract void accrueInterest() throws RemoteException;

}